napari_animation/interpolation.py

```diff
@@ -0,0 +1,91 @@
+import numbers
+from enum import Enum
+from functools import partial
+
+import numpy as np
+from napari._vispy.quaternion import quaternion2euler
+from scipy.spatial.transform import Rotation as R
+from scipy.spatial.transform import Slerp
+from vispy.util.quaternion import Quaternion
```

Comment:
I guess this could be ok for now, but I'd prefer if we didn't need the direct vispy import here. Maybe there is quaternion support inside scipy, or we can just vendor/ define the functions/ objects we need in a utils file here.

The reason is one day napari might support other rendering engines and so I'd like to keep our coupling to vispy as light as possible. This seems like one place where we could have a vispy independent alternative